net/dns/resolver/tsdns.go

```diff
@@ -650,14 +652,28 @@ func (r *Resolver) parseViaDomain(domain dnsname.FQDN, typ dns.Type) (netip.Addr
 
 	var siteID string
 	var ip4Str string
-	if strings.HasPrefix(fqdn, "via-") {
+	switch {
+	case strings.Contains(fqdn, "-via-"):
+		// Format number 3: "192-168-1-2-via-7" or "192-168-1-2-via-7.foo.ts.net."
+		// Third time's a charm. The earlier two formats follow after this block.
+		firstLabel, domain, _ := strings.Cut(fqdn, ".") // "192-168-1-2-via-7"
+		if !(domain == "" || dnsname.HasSuffix(domain, "ts.net") || dnsname.HasSuffix(domain, "tailscale.net")) {
+			return netip.Addr{}, false
```

Comment:
One unfortunate side-effect (which also happened with the previous name format) is that an inadvertent hostname matching like `netflix-via-europe.example.com` can't be resolved, we don't fallback to the default case.

Reply:
I don't think that's true. If this `parseViaDomain` returns false, we fall through and keep matching the old way.

```go
	// Special-case: 'via-<siteid>.<ipv4>' queries.
	if ip, ok := r.parseViaDomain(domain, typ); ok {
		return ip, dns.RCodeSuccess
	}
....
....
```

Reply:
Aha! Ok.

Reply:
Now that I see that comment, though, I'll update it, as that style is two styles old now.